From 6ff87f105f4e5eab11e8402974f3c756e5ce85fd Mon Sep 17 00:00:00 2001
From: Nathaniel Knight <nknight@cfenet.ubc.ca>
Date: Mon, 23 Dec 2019 15:18:34 -0800
Subject: [PATCH] Remove deprecated C-API macro

This commit replaces use of the deprecated
[`_PyOjbect_GC_UNTRACK`](https://docs.python.org/3.7/c-api/gcsupport.html#c._PyObject_GC_UNTRACK)
macro with calls to the equivalent
[`PyObject_GC_UnTrack`](https://docs.python.org/3.7/c-api/gcsupport.html#c.PyObject_GC_UnTrack)
function.

This change is required to support Python 3.8, which removes the
deprecated macro.
---
 SWIGWrappers/THyPhy_py3.cpp                      | 2 +-
 SWIGWrappers/THyPhy_python.cpp                   | 2 +-
 hyphy-src/src/lib/SWIGWrappers/THyPhy_py3.cpp    | 2 +-
 hyphy-src/src/lib/SWIGWrappers/THyPhy_python.cpp | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/SWIGWrappers/THyPhy_py3.cpp b/SWIGWrappers/THyPhy_py3.cpp
index 4ba7753..60bc6fc 100644
--- a/SWIGWrappers/THyPhy_py3.cpp
+++ b/SWIGWrappers/THyPhy_py3.cpp
@@ -3216,7 +3216,7 @@ SwigPyBuiltin_FunpackSetterClosure (PyObject *obj, PyObject *val, void *closure)
 
 SWIGINTERN void
 SwigPyStaticVar_dealloc(PyDescrObject *descr) {
-  _PyObject_GC_UNTRACK(descr);
+  PyObject_GC_UnTrack(descr);
   Py_XDECREF(PyDescr_TYPE(descr));
   Py_XDECREF(PyDescr_NAME(descr));
   PyObject_GC_Del(descr);
diff --git a/SWIGWrappers/THyPhy_python.cpp b/SWIGWrappers/THyPhy_python.cpp
index 1d2120a..6fdbff4 100644
--- a/SWIGWrappers/THyPhy_python.cpp
+++ b/SWIGWrappers/THyPhy_python.cpp
@@ -3216,7 +3216,7 @@ SwigPyBuiltin_FunpackSetterClosure (PyObject *obj, PyObject *val, void *closure)
 
 SWIGINTERN void
 SwigPyStaticVar_dealloc(PyDescrObject *descr) {
-  _PyObject_GC_UNTRACK(descr);
+  PyObject_GC_UnTrack(descr);
   Py_XDECREF(PyDescr_TYPE(descr));
   Py_XDECREF(PyDescr_NAME(descr));
   PyObject_GC_Del(descr);
diff --git a/hyphy-src/src/lib/SWIGWrappers/THyPhy_py3.cpp b/hyphy-src/src/lib/SWIGWrappers/THyPhy_py3.cpp
index 1d2120a..8312f1d 100644
--- a/hyphy-src/src/lib/SWIGWrappers/THyPhy_py3.cpp
+++ b/hyphy-src/src/lib/SWIGWrappers/THyPhy_py3.cpp
@@ -3216,7 +3216,7 @@ SwigPyBuiltin_FunpackSetterClosure (PyObject *obj, PyObject *val, void *closure)
 
 SWIGINTERN void
 SwigPyStaticVar_dealloc(PyDescrObject *descr) {
-  _PyObject_GC_UNTRACK(descr);
+  PyObject_GC_Untrack(descr)
   Py_XDECREF(PyDescr_TYPE(descr));
   Py_XDECREF(PyDescr_NAME(descr));
   PyObject_GC_Del(descr);
diff --git a/hyphy-src/src/lib/SWIGWrappers/THyPhy_python.cpp b/hyphy-src/src/lib/SWIGWrappers/THyPhy_python.cpp
index 1d2120a..6fdbff4 100644
--- a/hyphy-src/src/lib/SWIGWrappers/THyPhy_python.cpp
+++ b/hyphy-src/src/lib/SWIGWrappers/THyPhy_python.cpp
@@ -3216,7 +3216,7 @@ SwigPyBuiltin_FunpackSetterClosure (PyObject *obj, PyObject *val, void *closure)
 
 SWIGINTERN void
 SwigPyStaticVar_dealloc(PyDescrObject *descr) {
-  _PyObject_GC_UNTRACK(descr);
+  PyObject_GC_UnTrack(descr);
   Py_XDECREF(PyDescr_TYPE(descr));
   Py_XDECREF(PyDescr_NAME(descr));
   PyObject_GC_Del(descr);
